models/posts.js

```diff
@@ -0,0 +1,77 @@
+import yup from "yup";
+import { locSchema } from "./locations.js";
+
+const postTypes = {
+    OFFER: "offer",
+    REQUEST: "request"
+}
+
+const postCategories = {
+    TRANSPORT: "transport",
+    HOUSING: "housing",
+    SERVICES: "services",
+    GOODS: "goods",
+    FOOD: "food",
+    PET: "pet care",
+    CHILD: "child care",
+    ELDER: "elder care",
+    EDUCATION: "education",
+    EVENT: "event",
+    TOOL: "tool",
+    REPAIR: "repair",
+    LAND: "landcare",
+    OTHER: "other"
+}
+
+export const titleSchema = yup
+  .string()
+  .trim()
+  .min(5, "Title must be at least 5 characters long")
+  .max(100, "Title cannot exceed 100 characters")
+  .required("Title is required");
+
+export const contentSchema = yup
+  .string()
+  .trim()
+  .min(10, "Content must be at least 10 characters long")
+  .max(5000, "Content cannot exceed 5000 characters")
+  .required("Content is required");
+
+export const postTypeSchema = yup
+  .string()
```

Comment:
```suggestion
  .string()
  .trim()
```
might want to trim here too and for `postCategorySchema` just in case